File: CppPractice/CppLeetcode/Leetcode110.cpp
class Solution {
public:
    int helper(TreeNode *node)
    {
        if(!node){return 0;}
        int leftH = helper(node->left);
        int rightH = helper(node->right);
        if(leftH==-1||rightH==-1||abs(rightH-leftH)>1){return -1;}

        return 1+max(leftH,rightH);
    }
    bool isBalanced(TreeNode* root) {

        if(!root){return true;}
        return helper(root)!=-1;
        
    }
};

File: CSharpPractice/LeetcodePractice/Leetcode110.cs
public class Solution110 {
    int helper(TreeNode node)
    {
        if(node==null){return 0;}
        int rightH = helper(node.right);
        if(rightH==-1){return -1;}
        int leftH = helper(node.left);
        if(leftH==-1){return -1;}
        if(Math.Abs(rightH-leftH)>1){return -1;}
        return Math.Max(rightH,leftH)+1;
    }
    public bool IsBalanced(TreeNode root) 
    {
        if(root==null){return true;}
        return helper(root)!=-1;
    }
}